website/project/model.py

```diff
@@ -852,16 +856,48 @@ class Node(GuidStoredObject, AddonModelMixin, IdentifierMixin):
     # Primary institution node is attached to
     primary_institution = fields.ForeignField('institution')
 
+    affiliated_institutions = fields.ForeignField('institution', list=True)
+
     def add_primary_institution(self, user, inst):
-        if inst.auth(user):
-            self.primary_institution = inst
-            self.save()
-            return True
-        return False
+        if not inst.auth(user):
+            raise UserNotAffiliatedError('User is not affiliared with {}'.format(inst.name))
+        if inst == self.primary_institution:
+            return False
+        self.primary_institution = inst
+        if inst not in self.affiliated_institutions:
+            self.affiliated_institutions.append(inst)
+        self.add_log(
+            action=NodeLog.INSTITUTION_ADDED,
+            params={
+                'node': self._primary_key,
+                'institution': {
+                    'id': inst._id,
+                    'name': inst.name
+                }
+            },
+            auth=Auth(user)
+        )
+        return True
 
-    def remove_primary_institution(self):
+    def remove_primary_institution(self, user):
+        inst = self.primary_institution
+        if not inst:
+            return False
         self.primary_institution = None
-        self.save()
+        if inst in self.affiliated_institutions:
+            self.affiliated_institutions.remove(inst)
+        self.add_log(
```

Comment:
Same here--this log should only be created if the institution is actually removed.